### PR TITLE
Deep VM Queries (for older epochs, as well)

### PR DESCRIPTION
**Pull request by andreibancioiu**
## Reasoning behind the pull request

Enabled VM queries for older (all) epochs. Previously, VM queries were possible at blocks within epochs covered by the active persisters (e.g. last 3 epochs).

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Comment by andreibancioiu** [comment]
:ballot_box_with_check:  Tested using integration test: https://github.com/multiversx/mx-chain-go/pull/5935

:ballot_box_with_check:  Tested on Testnet:
1. sent transactions that alter the state of the smart contract, across multiple epochs
2. sent transactions the reveal the state root hash / block nonce etc. (at the moment of transaction processing)
3. performed deep queries to check the state at various blocks in the past - with respect to point (1)
4. performed deep queries to query the perceived "now" (state root hash, block nonce etc.) for blocks that held transactions from point (2)

Contract used to test on Testnet:
https://testnet-explorer.multiversx.com/accounts/erd1qqqqqqqqqqqqqpgq6xcct3kqdcsqndw2mmv29fqh9zq6yyp5396qszts9z

File: integrationTests/vm/wasm/queries/queries_test.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+	historyOfGetState[3] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 3})
+	historyOfGetNow[3] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 3})
+
+	historyOfGetState[4] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
+	historyOfGetNow[4] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
+
+	require.Equal(t, snapshotsOfGetState[1], historyOfGetState[1])
+	require.Equal(t, snapshotsOfGetNow[1].blockNonce, historyOfGetNow[1].blockNonce)
+	// This does not seem right!
```
Comment (andreibancioiu) [inline integrationTests/vm/wasm/queries/queries_test.go:117]: Though, I think the issue is with the test node, not with the actual implementation of deep queries. Actual implementation tested on system-test / Testnet :ballot_box_with_check:

**Review by sstanculeanu — APPROVED** [review]

File: factory/api/apiResolverFactory.go [hunk 2]
```diff
@@ -356,10 +356,20 @@ func createScQueryElement(
 		return nil, errDecode
 	}
 
+	apiBlockchain, err := blockchain.NewBlockChain(disabled.NewAppStatusHandler())
```
Comment (andreibancioiu) [inline factory/api/apiResolverFactory.go:359]: Here we should handle metachain, as well.

File: factory/api/export_test.go [hunk 3]
```diff
@@ -47,3 +48,7 @@ func CreateScQueryElement(args SCQueryElementArgs) (process.SCQueryService, erro
 		guardedAccountHandler: args.GuardedAccountHandler,
 	})
 }
+
+func CreateBlockchainForScQuery(selfShardID uint32) (data.ChainHandler, error) {
```
Comment (sstanculeanu) [inline factory/api/export_test.go:52]: missing comment

**Review by sstanculeanu — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.15-dev-config-eb2e06c06d -> sandbox-queries-dc797c7a20

--- Specific errors ---

block hash does not match 1448
wrong nonce in block 781
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  4
Nr. of all WARNS:  573
Nr. of new ERRORS:  0
Nr. of new WARNS:  4
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

upcloud-fra-validator-19 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  2

upcloud-fra-validator-29 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  2

/------/

File: factory/api/apiResolverFactory.go (hunk 2 above)
Comment (andreibancioiu) [reply factory/api/apiResolverFactory.go]: Fixed in https://github.com/multiversx/mx-chain-go/pull/5958.

File: integrationTests/vm/wasm/queries/queries_test.go (hunk 1 above)
Comment (andreibancioiu) [reply integrationTests/vm/wasm/queries/queries_test.go]: Removed that (redundant?) assertion.

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.15-dev-config-eb2e06c06d -> sandbox-queries-dc797c7a20

--- Specific errors ---

block hash does not match 1448
wrong nonce in block 781
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  4
Nr. of all WARNS:  573
Nr. of new ERRORS:  0
Nr. of new WARNS:  4
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

upcloud-fra-validator-19 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  2

upcloud-fra-validator-29 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  2

/------/

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**